File: org.kevoree.modeling.microframework/src/main/java/org/kevoree/modeling/api/time/impl/EntitiesMeta.java
package org.kevoree.modeling.api.time.impl;

import java.util.HashSet;

/**
 * Created by duke on 6/4/14.
 */

/*
 *  domainKey : path
  * */

public class EntitiesMeta {

    boolean isDirty = false;

    private static final String sep = "#";

    HashSet<String> list = new HashSet<String>();

    @Override
    public String toString() {
        StringBuilder stringBuilder = new StringBuilder();
        boolean isFirst = true;
        for (String p : list) {
            if (!isFirst) {
                stringBuilder.append(sep);
            }
            stringBuilder.append(p);
            isFirst = false;
        }
        return stringBuilder.toString();
    }

    public void load(String payload) {
        if (payload.equals("")) {
            return;
        }
        //TODO refactor this code for efficiency
        String[] elements = payload.split(sep);
        for (int i = 0; i < elements.length; i++) {
            list.add(elements[i]);
        }
        isDirty = false;
    }

}

File: org.kevoree.modeling.microframework/src/main/java/org/kevoree/modeling/api/time/blob/EntitiesMeta.kt
package org.kevoree.modeling.api.time.blob

import java.util.HashMap

/**
 * Created by duke on 6/4/14.
 */

/*
 *  key : path
  * */

class EntitiesMeta() {

    var isDirty = false;

    val sep = "#"

    var list = HashMap<String, Boolean>()

    override fun toString(): String {
        val stringBuilder = StringBuilder()
        var isFirst = true
        for (p in list.keySet()) {
            if (!isFirst) {
                stringBuilder.append(sep)
            }
            stringBuilder.append(p);
            isFirst = false;
        }
        return stringBuilder.toString()
    }

    fun load(payload: String) {
        if (payload.equals("")) {
            return;
        }
        //TODO refactor this code for efficiency
        val elements = payload.split(sep)
        for (elem in elements) {
            list.put(elem, true)
        }
        isDirty = false;
    }

}
